### ISSUE TITLE
Rune Knowledge

### ISSUE THREAD

**Issue by NickMcConnell** [NickMcConnell/FirstAgeAngband#50]
"There is an odd issue with rune knowledge; characters seem to forget and remember all known runes at random. It ranges from just not recognizing that you know all the runes of a piece of equipment (i.e. marking it with {??} even when you know everything) to temporarily forgetting runes that you do know (i.e. not knowing what's on a piece of equipment when you inspect it, even if you should know). The bug with forgetting seems like it's caused by something in town, either buying or selling. It seems to fix itself by saving and reloading the game." - MITZE

**Comment by NickMcConnell** [comment]
"I have a Cloak of Protection that is both too good and bugged; 27% rAcid, 80% rFire, and 20% rCold, and it has an additional rune that refuses to identify, even with scrolls of identify rune (unrelated to my troubles with spontaneously forgetting rune knowledge; I made sure that that issue wasn't in effect for the character before trying)." - MITZE

**Comment by NickMcConnell** [comment]
"Just sold an unidentified potion in town and bought it back identified (EDIT: Just confirmed that AFAICT it's selling an unknown flavor to the store that seems to trigger the bug), something in that seems to have caused me to forget specifically the % resists on equipment--but not pvals or binary flags, like sustains or protections.

"AFAICT it only applies when [I]nspecting equipment; the rune knowledge menu itself still claims you know the resist flags, and if you look at the character sheet it'll still show you your resist totals as they should be." - MITZE

**Comment by NickMcConnell** [comment]
Also "Not sure whether this is a bug or not; several rune effects seem to have multiple runes attached to them. Experience drain, for example is listed under both curse runes and other runes in the knowledge menu. I just found a helm of the shadow realm, where I didn't know the experience drain rune. Read a scroll of identify rune, learned it, but when I put on the helm to test it out, the game told me I learned the rune again after doing so. " - MITZE
